lib/elixir/test/elixir/registry_test.exs

```diff
@@ -526,6 +526,11 @@ defmodule RegistryTest do
     end
   end
 
+  test "child_spec/1 uses :name as :id" do
+    assert %{id: :custom_name} = Registry.child_spec([name: :custom_name])
+    assert %{id: Registry} = Registry.child_spec([])
```

Comment:
Should this fail as we require a name?

Reply:
@fishcakez it will fail later on, so I decided to not do the checking here. Do you think here would be better?

Reply:
@josevalim if there is more than one way to get to that checking outside this one, then I wouldn't do it here for consistency but if not then I could switch to pattern matching the `name` key

Reply:
You can't pattern match on keyword lists because then you become order dependent. For example: `[kind: :duplicate, name: Foo]` and `[kind: :duplicate, name: Foo]`.

Reply:
@josevalim then raise an exception 😄